src/DataGrid.tsx

```diff
@@ -279,6 +289,14 @@ function DataGrid<R, SR>({
     enableVirtualization
   });
 
+  const viewportColumns = useViewportColumns({
```

Comment:
Had to split into two hooks as we need to calculate colspan for all the rows in the viewport. This is needed to ensure a column that spans a column in the viewport is also visible